runtime/src/bank.rs

```diff
@@ -293,7 +293,7 @@ impl Bank {
         *self.parent.write().unwrap() = None;
 
         let squash_accounts_start = Instant::now();
-        for p in &parents {
+        for p in parents.iter().rev() {
```

Comment:
😮

Reply:
@rob-solana i am not sure if getting the slot values out of self.ancestors and sorting them will be better.

Reply:
I'm worried that they might not be the same...

squash() doesn't change bank.ancestors, does it?

Reply:
ah, it might add extra roots?

Reply:
dunno.  I'm trying to understand how ancestors is used.  a hashmap of slot->depth that's passed all the way down to accounts_index (which is **really** opaque) is hard for me to prove to myself that it's correct

Reply:
Write some tests :).  The roots table in the index does the magic part.